core/src/main/java/com/linecorp/armeria/server/metric/PrometheusExpositionService.java

```diff
@@ -85,11 +84,17 @@ public PrometheusExpositionService(CollectorRegistry collectorRegistry) {
 
     @Override
     protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final String accept = req.headers().get(HttpHeaderNames.ACCEPT);
+        final String format = TextFormat.chooseContentType(accept);
         final ByteArrayOutputStream stream = new ByteArrayOutputStream();
         try (OutputStreamWriter writer = new OutputStreamWriter(stream)) {
-            TextFormat.write004(writer, collectorRegistry.metricFamilySamples());
+            TextFormat.writeFormat(
+                format,
+                writer,
+                collectorRegistry.metricFamilySamples()
+            );
```

Comment:
nit: We could probably merge these into one line. 😉 
```suggestion
            TextFormat.writeFormat(format, writer, collectorRegistry.metricFamilySamples());
```